webapp/src/components/github_issue_selector/github_issue_selector.jsx

```diff
@@ -0,0 +1,140 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+
+import debounce from 'debounce-promise';
+import AsyncSelect from 'react-select/async';
+
+import {getStyleForReactSelect} from 'utils/styles';
+import Client from 'client';
+
+const searchDebounceDelay = 400;
+
+export default class GithubIssueSelector extends Component {
+    static propTypes = {
+        required: PropTypes.bool,
+        theme: PropTypes.object.isRequired,
+        onChange: PropTypes.func.isRequired,
+        error: PropTypes.string,
+        value: PropTypes.object,
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.state = {invalid: false};
+    }
+
+    componentDidUpdate(prevProps, prevState) {
+        if (prevState.invalid && this.props.value !== prevProps.value) {
+            this.setState({invalid: false}); //eslint-disable-line react/no-did-update-set-state
+        }
+    }
+
+    handleIssueSearchTermChange = (inputValue) => {
+        return this.debouncedSearchIssues(inputValue);
+    };
+
+    searchIssues = (text) => {
+        const textEncoded = encodeURIComponent(text.trim().replace(/"/g, '\\"'));
+
+        return Client.searchIssues(textEncoded).then((data) => {
+            return Array.isArray(data) ? data.map((item) => ({value: item, label: item.title})) : [];
+        }).catch((e) => {
+            this.setState({error: e});
+        });
+    };
+
+    debouncedSearchIssues = debounce(this.searchIssues, searchDebounceDelay);
+
+    onChange = (e) => {
+        const value = e ? e.value : '';
+        this.props.onChange(value);
+    }
+
+    isValid = () => {
+        if (!this.props.required) {
+            return true;
+        }
+
+        const valid = this.props.value && this.props.value.toString().length !== 0;
+        this.setState({invalid: !valid});
+        return valid;
+    };
+
+    render = () => {
+        const {error} = this.props;
+        const requiredStar = (
+            <span
+                className={'error-text'}
+                style={{marginLeft: '3px'}}
+            >
+                {'*'}
+            </span>
+        );
+
+        let issueError = null;
+        if (error) {
+            issueError = (
+                <p className='help-text error-text'>
+                    <span>{error}</span>
+                </p>
+            );
+        }
+
+        const serverError = this.state.error;
+        let errComponent;
+        if (this.state.error) {
+            errComponent = (
+                <p className='alert alert-danger'>
+                    <i
+                        className='fa fa-warning'
+                        title='Warning Icon'
+                    />
+                    <span> {serverError.toString()}</span>
```

Comment:
Nit: unnecessary whitespace here
```suggestion
                    <span>{serverError.toString()}</span>
```

Reply:
done